src/ProxyWrappers.cxx

```diff
@@ -371,37 +403,7 @@ static int BuildScopeProxyDict(Cppyy::TCppScope_t scope, PyObject* pyclass, cons
  // collect data members (including enums)
     std::vector<Cppyy::TCppScope_t> datamembers = Cppyy::GetDatamembers(scope);
     for (auto &datamember : datamembers) {
-    // allow only public members
-        if (!Cppyy::IsPublicData(datamember))
-            continue;
-
-    // enum datamembers (this in conjunction with previously collected enums above)
-        if (Cppyy::IsEnumType(Cppyy::GetDatamemberType(datamember)) && Cppyy::IsStaticDatamember(datamember)) {
-        // some implementation-specific data members have no address: ignore them
-            if (!Cppyy::GetDatamemberOffset(datamember))
-                continue;
-
-        // two options: this is a static variable, or it is the enum value, the latter
-        // already exists, so check for it and move on if set
-            PyObject* eset = PyObject_GetAttrString(pyclass,
-                const_cast<char*>(Cppyy::GetFinalName(datamember).c_str()));
-            if (eset) {
```

Comment:
The reason behind moving the entire logic into `AddPropertyToClass` is to be able to use recursion for nested anonymous `struct`s and `union`s.
Test cases do not fail, so the refactor is safe.

Let me know if there is a specific change that looks off to you.

Reply:
> I was wondering how masters support nested anonymous structs and unions without recursion?

I did not have a look at it. I saw the problem and came up with a solution. The solution looked simple and easy to implement. So I did not look at how master does this.

Reply:
I understand that but one of the main points here is not to diverge from master as much as possible. We can make as many changes in clingwrapper and add interfaces in InterOp to facilitate/emulate what is made available to CPyCppyy to achieve the same result. Since the support does not use recursion in master and is achieved with the interfaces at hand, we would recommend a side-by-side debug to emulate the same on the backend side.

Reply:
Yes, eventually we should merge all of this work back to upstream and if that change is not mergeable will be a problem. I cannot judge that now but perhaps we could discuss that at the meeting later today.